sample/src/main/java/com/massivedisaster/adal/sample/feature/connectivity/FragmentConnectivityAware.java

```diff
@@ -0,0 +1,127 @@
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+import android.widget.TextView;
+
+import com.massivedisaster.adal.connectivity.ConnectionChangeReceiver;
+import com.massivedisaster.adal.connectivity.NetworkUtils;
+import com.massivedisaster.adal.fragment.BaseFragment;
+import com.massivedisaster.adal.sample.R;
+
+/**
+ * Connectivity Change Fragment meant to test the {@link ConnectionChangeReceiver} for connectivity changes.
+ */
+public class FragmentConnectivityAware extends BaseFragment {
+
+    private TextView mTxtMessage;
+
+    /**
+     * Network verification.
+     */
+    private ConnectionChangeReceiver mConnectionReceiver = new ConnectionChangeReceiver() {
+        @Override public void onReceive(@NonNull final Context context, @NonNull final Intent intent) {
+            if (ConnectionChangeReceiver.CONNECTIVITY_CHANGE_FILTER.equals(intent.getAction())) {
+                checkConnectivity();
+            }
+        }
+    };
+
+    @Override protected void getFromBundle(Bundle bundle) {
+        // Intended.
+    }
+
+    @Override protected int layoutToInflate() {
+        return R.layout.fragment_connectivity_aware;
+    }
+
+    @Override protected void restoreInstanceState(@Nullable Bundle savedInstanceState) {
+        // Intended.
+    }
+
+    @Override protected void doOnCreated() {
+        getActivity().setTitle(R.string.sample_network);
+        mTxtMessage = findViewById(R.id.txtMessage);
+    }
+
+    /**
+     * Called when FragmentConnectivityAware is about to become visible.
+     */
+    @Override public void onStart() {
+        super.onStart();
+        mConnectionReceiver.registerConnectionChangeReceiver(getActivity());
+    }
+
+    /**
+     * Called when the FragmentConnectivityAware has become visible.
+     */
+    @Override public void onResume() {
+        super.onResume();
+        checkConnectivity();
+    }
+
+    /**
+     * Called just before the FragmentConnectivityAware is destroyed.
+     */
+    @Override public void onDestroy() {
+        if (mConnectionReceiver != null) {
+            mConnectionReceiver.unregisterConnectionChangeReceiver();
+        }
+        super.onDestroy();
+    }
+
+    /**
+     * Check whether the device has Internet connectivity or not.
+     */
+    private void checkConnectivity() {
+        if (!getActivity().isFinishing() && isVisible()) {
+            final boolean isOnline = NetworkUtils.isNetworkConnected(getActivity());
+            handleConnectivityStatusChange(isOnline);
+        }
+    }
+
+    /**
+     * Log the device Internet connectivity status.
+     *
+     * @param isOnline boolean value indicating whether the device has a connection established or not.
+     */
+    private void handleConnectivityStatusChange(final boolean isOnline) {
```

Comment:
@paribeiro10 This one its just a enhancement. You could remove the "duplicated" lines here, adding a ternary operator to get the string: 

```
String deviceConnectivity = isOnline ? getString(R.string.connectivity_device_online) : getString(R.string.connectivity_device_offline);

mTxtMessage.setText(Connectivity);
Log.d(getActivity().getClass().getName(), Connectivity);
```

Plus, the `Log` here seems irrelevant.

Reply:
Good catch. Replaced.